src/DynamoCore/Models/PresetModel.cs

```diff
@@ -98,56 +81,34 @@ public PresetModel(string name, string description, IEnumerable<NodeModel> input
         //this overload is used for loading
         private PresetModel(string name, string description, List<NodeModel> nodes, List<XmlElement> serializedNodes, Guid id)
         {
-
             Name = name;
             Description = description;
             this.nodes = nodes;
             this.serializedNodes = serializedNodes;
-            //if we have not supplied a guid at load then create a new one
-            if (id == Guid.Empty)
-            {
-                guid = Guid.NewGuid();
-            }
-            else
-            {
-                guid = id;
-            }
-        }
-        #endregion
-
-
-
-        #region serialization / deserialzation
-
-        //grabbed some methods needed from modelbase for serialization
-        protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext context)
-        {
-            string typeName = GetType().ToString();
-            XmlElement element = xmlDocument.CreateElement(typeName);
-            return element;
+            GUID = id;
         }
 
         /// <summary>
-        /// we serialze the presets to xml like a model, but we deserialze them before the workspacemodel is constructed
-        /// during save of the graph, workspace model serializes its list of presetModels into a "Presets" xml element
-        /// when a new presetModel is created we'll serialize all the current nodes into a new xmlelement
-        /// but not actually write this xml to a file until the graph is saved
+        /// this overload is used for loading with deserializeCore, we must pass the nodesInTheGraph to the instance of the Preset so that
+        /// we can detect missing nodes
         /// </summary>
-        /// <param name="xmlDocument"></param>
-        /// <param name="context"></param>
-        /// <returns></returns>
-        public XmlElement Serialize(XmlDocument xmlDocument, SaveContext context)
+        /// <param name="nodesInGraph"></param>
+ 
+        public PresetModel(IEnumerable<NodeModel> nodesInGraph)
         {
-            var element = CreateElement(xmlDocument, context);
-            SerializeCore(element, context);
-            return element;
+            this.nodes = nodesInGraph.ToList();
```

Comment:
at construction time when loading a preset store all the nodes in the current node graph in the nodes field, then when it's deserialized we replace this with the nodes that the preset stores.